OpenRA.Mods.Common/Traits/Render/DrawLineToTarget.cs

```diff
@@ -72,13 +72,18 @@ public IEnumerable<IRenderable> RenderAfterWorld(WorldRenderer wr)
 			if (targets == null || targets.Count == 0)
 				yield break;
 
+			List<WPos> valid_targets = new List<WPos>();
+			valid_targets.Add(self.CenterPosition);
+
 			foreach (var target in targets)
 			{
 				if (target.Type == TargetType.Invalid)
 					continue;
 
-				yield return new TargetLineRenderable(new[] { self.CenterPosition, target.CenterPosition }, c);
+				valid_targets.Add(target.CenterPosition);
 			}
+
+			yield return new TargetLineRenderable(valid_targets, c);
```

Comment:
Why are we even using yield returns and yield breaks here if we aren't returning multiples inside the foreach loop?
Please use a normal return and return an empty array in cases where yield break is as yield is a pretty expensive keyword that shouldn't be used if there is no reason to.